### ISSUE TITLE
Problem with automatic deploys, data-only containers and volume masking warning.

### ISSUE THREAD

**Issue by do3cc** [docker/compose#2465]
This is a terrible subject but I am not sure how to phrase it better.
I am using docker-compose, together with data-only containers.
This is my relevant part:
```
solr:
  image: some_solr
  volumes_from:
    - solr_data
solr_data:
  image: some_solr
  volumes:
    - /var/solr:/var/solr:z
```
My CI Server logs in, does a docker pull for all images, then runs docker-compose up.

I expected that all containers based on old images get deleted and recreated and everything is fine.

Instead I get a warning that now my external volumes aren't used any more.

I am now not sure,
- Is the warning wrong?
- Should this be better handled by docker-compose up so that masking won't happen
- Should I declare some dependency between the two containers to avoid this
- Should I rm the containers?

**Comment by dnephin** [comment]
The warning may be wrong. There is a fix in #2390 coming in 1.5.2 to fix incorrect warning in some cases. You can try the fix from the master build: https://dl.bintray.com/docker-compose/master/

**Comment by dnephin** [comment]
I believe this was fixed in Compose 1.5.2. Were you able to upgrade and try it out?

**Comment by do3cc** [comment]
Sorry for the late reply, yest 1.5.2 fixed it
